lib/index.js

```diff
@@ -29,15 +29,11 @@ export default postcss.plugin('postcss-map', opts => {
       readFile(map, 'utf-8', (err, data) => {
         if (err)
           return reject(err);
-
-        try {
-          let name = path.basename(map, path.extname(map));
-          maps[name] = yaml.safeLoad(data);
-          resolve();
-        } catch (e) {
-          reject(e);
-        }
+        resolve(data);
       });
+    }).then(function (data) {
+      let name = path.basename(map, path.extname(map));
+      maps[name] = yaml.safeLoad(data);
```

Comment:
As I said all methods implement catcing all expections inside them. Example of realization
https://github.com/floatdrop/pinkie/blob/master/index.js#L58-L63
Native do the same.